### ISSUE TITLE
Async image loading - need for componentWillUnmount

### ISSUE THREAD

**Issue by kriskate** [archriss/react-native-render-html#229]
### Is this a bug report or a feature request?

bug report

### Have you read [the guidelines](https://github.com/archriss/react-native-render-html/blob/master/CONTRIBUTING.md) regarding bug report?

Yes

### Have you read [the documentation](https://github.com/archriss/react-native-render-html/blob/master/README.md) in its entirety?

Yes

### Have you made sure that your issue hasn't already been reported/solved?

Yes

### Is the bug specific to iOS or Android? Or can it be reproduced on both platforms?

Both

### Is the bug reproductible in a production environment (not a debug one)?

Yes

### Have you been able to reproduce the bug in [the provided example](https://github.com/archriss/react-native-render-html/tree/master/Demo)?

To be honest, haven't tried, hear me out...

### Environment

  Environment:
    React: 16.5.0
    React native: 0.57
    react-native-render-html: 3.10.0

  Target Platform:
    Android (multiple)
    iOS (12/ 10.3.3)
  

### Steps to Reproduce

1. have an RenderHTML with multiple images that occupies the JSThread
2. before all the images are loaded, the user is able to perform actions between image loads (ie unmount RenderHTML)
3. React throws a warning because RenderHTML tries to setState on an unmounted component

### Expected Behavior

The not-yet-fully-rendered component unmounts correctly.

### Actual Behavior

RenderHTML tries to set its state after it's unmounted.

```
"Warning: Can't call setState (or forceUpdate) on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.%s, "
```

### Fix

I could submit a pull request for this, but there might be a better fix I don't know about.

1. [HTML.js](https://github.com/archriss/react-native-render-html/blob/master/src/HTML.js#L111) needs a state of "mounted"
2. at componentWillUnmount, set state.mounted to false
3. in registerDOM, after the async call:
```
let response = await fetch(uri);
if(state.mounted) return undefined;
```

### Demo

I'll provide a reproducible demo next week

### LE (Fix # 2):
Because of React's synchronous nature, it's impossible to setState before the async workload is done, in some cases (haven't tested on ```react-native-render-html```).

What I ended up doing in a similar situation is setting a variable in the component file, where I'm doing async fetches, and just modifying that var at componentWillUnmount.
Not a best practice, but it gets the job done, I guess...

In Html.js's case, this would translate to something like:

```
let isCancelled = false;

export default class HTML extends PureComponent {
[...]
    componentWillUnmount() {
        isCancelled = true;
    }
    async registerDOM (props = this.props, cb) {
        [...]
       
                    let response = await fetch(uri);
                    !isCancelled && this.setState({ dom: response._bodyText, loadingRemoteURL: false });
        [...]
    }
}
```

**Comment by kriskate** [comment]
As promised, I made a [snack](https://snack.expo.io/@kriskate/react-native-render-html-async-unmounting). 
Fortunately, I cannot reproduce it, thus I've updated the react-native-render-html dependency in my project to @4.1.1 and it does not seem to happen anymore.

Thank you!

**Comment by Exilz** [comment]
Glad to know this is working.
If anyone is reading this, this has been fixed since `4.1.0` 👍 4fca299
